File: src/Tiantong.Wms.Api/Entities/Order.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using Newtonsoft.Json;

namespace Tiantong.Wms.Api
{
  [Table("orders")]
  public class Order : Entity
  {
    public virtual int warehouse_id { get; set; }

    public virtual string number { get; set; }

    public virtual int operator_id { get; set; }

    public virtual int applicant_id { get; set; }

    public virtual int department_id { get; set; }

    public virtual int supplier_id { get; set; }

    [JsonIgnore]
    public virtual string type { get; set; }

    public virtual string status { get; set; } = "";

    public virtual string comment { get; set; } = "";

    public virtual DateTime due_time { get; set; } = DateTime.MinValue;

    public virtual DateTime created_at { get; set; } = DateTime.Now;

    public virtual DateTime finished_at { get; set; } = DateTime.MinValue;

    [ForeignKey("order_id")]
    public virtual List<OrderPayment> payments { get; set; }

    [ForeignKey("order_id")]
    public virtual List<OrderItem> items { get; set; }
  }
}

File: src/Tiantong.Wms/Tiantong.Wms.Client/src/Entities/Order.ts
import OrderItem from './OrderItem'
import OrderPayment from './OrderPayment'
import { DateTime } from '@/utils/common'

export default class PurchaseOrder {
  id: number = 0

  warehouse_id: number = 0

  number: string | null = null

  operator_id: number = 0

  applicant_id: number = 0

  department_id: number = 0

  supplier_id: number = 0

  status: string = 'created'

  comment: string = ''

  due_time: string = DateTime.now

  created_at: string = DateTime.now

  finished_at: string = DateTime.now

  items: Array<OrderItem>

  payments: Array<OrderPayment>

  constructor (
    payments: Array<OrderPayment> = [],
    items: Array<OrderItem> = [],
  ) {
    this.items = items
    this.payments = payments
  }
}
